lib/streamlit/state/widgets.py

```diff
@@ -182,59 +218,102 @@ def coalesce_widget_states(
     return coalesced
 
 
-class WidgetManager:
+class WidgetManager(object):
     """Stores widget values for a single connected session."""
 
     def __init__(self):
```

Comment:
We could (optionally) initialize the widget manager from WidgetStates, which would simplify some other parts of the code and move towards a slightly more functional style

Reply:
Will need a bit more clarification on what you mean here. In the interest of time, I think the refactor is something we'll probably want to hold off on until post-hackathon

Reply:
I meant that all classes implicitly inherit from `object`, so you can omit it and just write `class Foo():` or `class Foo:` instead of `class Foo(object):`

Reply:
I meant that instead of initializing a WidgetManager with empty state dicts, and then immediately call `set_state` or whatever it is called, we could have `__init__` take an optional WidgetState argument and fill in the dict right away (even by just calling the same method other could would have). That way, other code wouldn't need to worry about remembering to call the function to imperatively set the state, in the common case where the calling code already has the state it wants to put in the manager.